src/ui/save-html.js

```diff
@@ -43,139 +80,16 @@ if (supportsDownload) {
-    url: document.location.href,
-  });
-  return `${url}?${params}`;
-}
-
-function cleanup(cloneDoc) {
-  const { head, body, documentElement } = cloneDoc;
-  cleanupHyper(documentElement);
-  Array.from(
-    cloneDoc.querySelectorAll(".removeOnSave, #toc-nav")
-  ).forEach(elem => elem.remove());
-  body.classList.remove("toc-sidebar");
-  removeReSpec(documentElement);
-  const insertions = cloneDoc.createDocumentFragment();
-  // Move meta viewport, as it controls the rendering on mobile.
-  const metaViewport = cloneDoc.querySelector("meta[name='viewport']");
-  if (metaViewport && head.firstChild !== metaViewport) {
-    insertions.appendChild(metaViewport);
-  }
-  // Move charset to near top, as it needs to be in the first 512 bytes.
-  let metaCharset = cloneDoc.querySelector(
-    "meta[charset], meta[content*='charset=']"
-  );
-  if (!metaCharset) {
-    metaCharset = cloneDoc.createElement("meta");
-    metaCharset.setAttribute("charset", "utf-8");
-  }
-  insertions.appendChild(metaCharset);
-  // Add meta generator
-  const metaGenerator = cloneDoc.createElement("meta");
-  metaGenerator.name = "generator";
-  metaGenerator.content =
-    "ReSpec " + window.respecVersion || "Developer Channel";
-  insertions.appendChild(metaGenerator);
-  head.insertBefore(insertions, head.firstChild);
-  pub("beforesave", documentElement);
-}
-
-function cleanupHyper(node) {
-  // collect first, or walker will cease too early
-  const comments = [...walkTree(
-    node, NodeFilter.SHOW_COMMENT, 
-    comment => comment.textContent.startsWith("_hyper")
-  )];
-  for (const comment of comments) {
-    comment.parentNode.removeChild(comment);
-  }
-}
-
-function* walkTree(...args) {
-  const walker = document.createTreeWalker(...args);
-  while (walker.nextNode()) {
-    yield walker.currentNode;
-  }
-}
-
-function toDownloadLink(
-  { id, href, fileName, title, type = "" } = { type: "" }
-) {
-  const a = document.createElement("a");
-  a.classList.add("respec-save-button");
-  a.id = id;
-  a.href = href;
-  a.download = fileName;
-  a.type = type;
-  a.addEventListener("click", () => {
-    ui.closeModal();
-  });
-  const render = hyperHTML.bind(a);
-  return render`
-    ${title}
-  `;
+export function show() {
+  if (!supportsDownload) return;
+  saveDialog.show(button);
 }
 
-const div = document.createElement("div");
-div.classList.add("respec-save-buttons");
-const dialogRender = hyperHTML.bind(div);
-const saveDialog = {
-  show(button) {
-    document.respecIsReady.then(() => {
-      dialogRender`
-        ${downloadLinks.map(toDownloadLink)}
-      `;
-      ui.freshModal(l10n[lang].save_snapshot, div, button);
-    });
-  },
-};
-
 export function exportDocument(format, mimeType) {
-  // Bug in babel won't export async function for some reason?
-  return document.respecIsReady.then(() => {
-    const dataURL = toDataURL(serialize(format), mimeType);
-    const encodedString = dataURL.replace(/^data:\w+\/\w+;charset=utf-8,/, "");
```

Comment:
Whoops, we missed this bit.

Reply:
oops. my bad. sorry about this. hopefully not too much damage

Reply:
Not too much :) we took out the ability to generate specs for a little while. We have add a small test for this. Basically, we need to check if the exported file starts with `<html>`.

Reply:
(also, not your fault... we both worked on it together, and I also missed it in review - it happens and doing hot fixes at 1:30am is what I live for 🤓🤘)